src/WinGet.RestSource.PowershellSupport/YamlToRestConverter.cs

```diff
@@ -39,14 +40,10 @@ public static class YamlToRestConverter
                             Path.GetRandomFileName() + ".yaml");
             if (packageFiles.Length > 1)
             {
-                // Multi File manifest case
-                // The winget client ValidateManifest function
-                // merges the manifests into the merged manifest format.
-                (bool succeeded, _) = WinGetUtilWrapperManifest.ValidateManifest(
-                    directory,
-                    mergedManifestFilePath);
+                var factory = new WinGetFactory();
+                using var manifestResult = factory.CreateManifest(directory, mergedManifestFilePath, WinGetCreateManifestOption.SchemaAndSemanticValidation);

```

Comment:
We should probably revisit code here when we work on making the code production ready. I see in multifile case, we are validating the manifest, while in singleton or merged case, we did not validate manifest (the else block below).